modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/CheckpointCommand.java

```diff
@@ -46,13 +51,24 @@ public class CheckpointCommand extends AbstractCommand<Void> {
         return CHECKPOINT.toCommandName();
     }
 
+    /** ID of a node to run checkpoint at */
+    private UUID nodeId;
+
     /** {@inheritDoc} */
     @Override public Object execute(GridClientConfiguration clientCfg, Logger log) throws Exception {
         try (GridClient client = Command.startClient(clientCfg)) {
             GridClientCompute compute = client.compute();
 
             // Try to find connectable server nodes
-            Collection<GridClientNode> nodes = compute.nodes((n) -> n.connectable() && !n.isClient());
+            Collection<GridClientNode> nodes;
+
+            if (nodeId == null)
+                nodes = compute.nodes((n) -> n.connectable() && !n.isClient());
+            else {
+                nodes = new ArrayList<>();
+
+                nodes.add(compute.node(nodeId));
```

Comment:
If the collection will no longer mutate, then it is better to use `java.util.Arrays#asList`.